integration/securitypolicy_test.go

```diff
@@ -0,0 +1,400 @@
+					Spec: v1alpha1.SecurityPolicySpec{
+						ID:     spec.MID,
+						Active: true,
+						State:  initialK8sPolicyState,
+						Rate:   initialK8sPolicyRate,
+						Tags:   []string{initialK8sPolicyTag},
+					},
+				}
+
+				_, err = util.CreateOrUpdate(ctx, polRec.Client, &policyCR, func() error {
+					return nil
+				})
+				eval.NoErr(err)
+
+				err = wait.For(func() (done bool, err error) {
+					_, err = polRec.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(&policyCR)})
+					return err == nil, err
+				})
+				eval.NoErr(err)
+
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
+						policyOnK8s, ok := object.(*v1alpha1.SecurityPolicy)
+						eval.True(ok)
+						eval.True(len(policyOnK8s.Status.PolID) > 0)
+
+						policyOnTyk, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						eval.NoErr(err)
+
+						eval.True(policyOnK8s.Spec.MID == policyOnTyk.MID &&
+							len(policyOnK8s.Spec.Tags) == len(policyOnTyk.Tags) &&
+							len(policyOnTyk.Tags) == 1 &&
+							policyOnTyk.Tags[0] == initialK8sPolicyTag &&
+							policyOnTyk.Rate == initialK8sPolicyRate &&
+							policyOnTyk.State == initialK8sPolicyState)
+						return true
+					}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("Create a drift between k8s and Dashboard",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				// In order to create a drift between k8s and Dashboard, first delete the SecurityPolicy controlled
+				// by Operator from Dashboard. Now, in the next reconciliation, Operator should recreate same
+				// SecurityPolicy on Tyk Side based on the SecurityPolicy CR on k8s.
+
+				eval := is.New(t)
+
+				// Delete an existing Policy from Dashboard to create drift between Tyk and K8s state.
+				err := klient.Universal.Portal().Policy().Delete(reqCtx, policyCR.Status.PolID)
+				eval.NoErr(err)
+
+				// Ensure that policy is deleted from Tyk.
+				_, err = klient.Universal.Portal().Policy().Get(reqCtx, policyCR.Status.PolID)
+				eval.True(tykClient.IsNotFound(err))
+
+				// After reconciliation, Operator should detect drift and recreate non-existing policy on Tyk side.
+				err = wait.For(func() (done bool, err error) {
+					_, err = polRec.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(&policyCR)})
+					return err == nil, err
+				})
+				eval.NoErr(err)
+
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
+						policyOnK8s, ok := object.(*v1alpha1.SecurityPolicy)
+						eval.True(ok)
+
+						newSpec, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						eval.NoErr(err)
+
+						eval.True(hasSameValues(&policyOnK8s.Spec, newSpec, policyOnK8s.Status.PolID))
```

Comment:
Can you also check if the new and old(before deleting it from Dashboard) policyID are same?

Reply:
I compared previous & new IDs. Can you please re-check again (d17880a1cd921024b5b82813d7c62c2021ebb2cd) if it looks okay?

Reply:
Ohh right.